bluepyemodel/emodel_pipeline/plotting_utils.py

```diff
@@ -258,24 +281,73 @@ def fill_in_IV_curve_evaluator(evaluator, efel_settings, prot_name="iv"):
                 else:
                     prot_v_deflection.append(prot.name)
 
+    if new_amps is not None:
+        for protocol_name in prot_v_deflection + prot_max_v:
+            for amp in new_amps:
+                protocol_name_amp = f"{protocol_name.split('_')[0]}_{amp}"
+                if 0 <= amp < 100:
+                    prot_max_v.append(protocol_name_amp)
+                elif amp < 0:
+                    prot_v_deflection.append(protocol_name_amp)
+
     # maps protocols of interest with all its associated features
     # also get protocol data we need for feature registration
     prots_to_feats = {}
     prots_data = {}
-    for objective in evaluator.fitness_calculator.objectives:
-        feat = objective.features[0]
-        for protocol_name in prot_v_deflection + prot_max_v:
-            if protocol_name in feat.recording_names[""]:
+
+    for protocol_name in prot_v_deflection + prot_max_v:
+        matched_feat, feat_already_present = find_matching_feature(evaluator, protocol_name)
+        if matched_feat is not None:
+            if feat_already_present:
                 if protocol_name not in prots_to_feats:
                     prots_to_feats[protocol_name] = []
                 if protocol_name not in prots_data:
                     prots_data[protocol_name] = {
-                        "stimulus_current": feat.stimulus_current,
-                        "stim_start": feat.stim_start,
-                        "stim_end": feat.stim_end,
+                        "stimulus_current": matched_feat.stimulus_current,
+                        "stim_start": matched_feat.stim_start,
+                        "stim_end": matched_feat.stim_end,
                     }
-                prots_to_feats[protocol_name].append(feat.efel_feature_name)
-                continue
+                prots_to_feats[protocol_name].append(matched_feat.efel_feature_name)
+            else:
+                p_rel_name = matched_feat.recording_names[""].split(".")[0]
+                amp_rel = float(protocol_name.split("_")[1])
+                amp = float(matched_feat.recording_names[""].split(".")[0].split("_")[-1])
+                p_rel = updated_evaluator.fitness_protocols["main_protocol"].protocols[p_rel_name]
+                stimuli = [
```

Comment:
This bit looks like what you have in the update_evaluator function. Maybe you could create a function doing this and calling it instead of duplicating the code?

Reply:
done in the last commit